src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -113,27 +109,92 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
+        :targetPFN: str, destination PFN
+        :options: str, additional options for gfal-cp
+        :checksums: dict, collect checksums according to the algorithms saved as keys
+        """
+
+
+        copyCommandDict = self.buildCopyCommandDict(sourcePFN, targetPFN, options, checksums)
         copyCommand = self.copyCommand % copyCommandDict
-        result += copyCommand
+        result = "#!/bin/bash\n" + copyCommand
 
         if _CheckExitCodeOption:
             result += """
             EXIT_STATUS=$?
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
-               echo "ERROR: gfal-copy exited with $EXIT_STATUS"
-               echo "Cleaning up failed file:"
-               %s
+                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+                echo "Cleaning up failed file: {remove_command}"
             fi
             exit $EXIT_STATUS
-            """ % self.createRemoveFileCommand(targetPFN)
-
+            """.format(
+                remove_command=self.createRemoveFileCommand(targetPFN)
+            )
+        
         return result
 
+    def createDebuggingCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+        """
+        Debug a failed gfal-cp command for stageOut, without re-running it,
+        providing information on the environment and the certifications
+
+        :sourcePFN: str, PFN of the source file
+        :targetPFN: str, destination PFN
+        :options: str, additional options for gfal-cp
+        :checksums: dict, collect checksums according to the algorithms saved as keys
+        """
+        
+        copyCommandDict = self.buildCopyCommandDict(sourcePFN, targetPFN, options, checksums)
+        copyCommand = self.copyCommand % copyCommandDict
+
+        proxyInfo = None
+        try:
+            stdout, stderr, returnCode = runCommand("voms-proxy-info", timeout=10)
+            if returnCode == 0:
+                proxyInfo = stdout
+            else:
+                logging.debug("voms-proxy-info failed with return code: %s", returnCode)
+                logging.debug("error output: %s", stderr)
+        except SubprocessAlgoException as e:
+            logging.debug("An exception occurred while running voms-proxy-info:")
+            logging.debug(e)
+
+        logging.debug("Actual gfal-cp command which failed: %s", copyCommand)
+
+        result = "#!/bin/bash\n"
+        result += """
+        echo "gfal-copy command which failed: {copy_command}"
+        echo "Source PFN: {source}"
+        echo "Target PFN: {destination}"
+        echo "Proxy and JOBSTARTDIR: {setup_info}"
+        echo "Hostname:   $(hostname -f)"
+        echo "OS:  $(uname -n -r -s)
+        """.format(
+            copy_command=copyCommand,
+            source=copyCommandDict['source'],
+            destination=copyCommandDict['destination'],
+            setup_info=self.setups
+        )
+        if proxyInfo is not None:
```

Comment:
I feel like just printing the proxyInfo content in the try/except above will give log viewers a much better experience than passing it over (the actual content already retrieved) to a sub-process to execute all together. Especially if that can fail, because then it will print the whole command that failed